internal/clients/api_client.go

```diff
@@ -307,5 +346,87 @@ func newEsApiClient(d *schema.ResourceData, key string, version string, useEnvAs
 		es.Transport = newDebugTransport("elasticsearch", es.Transport)
 	}
 
-	return &ApiClient{es, nil, version}, diags
+	return es, diags
+}
+
+func buildKibanaClient(d *schema.ResourceData, baseConfig BaseConfig, useEnvAsDefault bool) (*kibana.Client, diag.Diagnostics) {
+	var diags diag.Diagnostics
+
+	kibConn, ok := d.GetOk("kibana")
+	if !ok {
+		return nil, diags
+	}
+
+	config := kibana.Config{}
+
+	// Use ES details by default
+	config.Username = baseConfig.Username
+	config.Password = baseConfig.Password
+	config.Address = baseConfig.Addresses[0]
+
+	// if defined, then we only have a single entry
+	if kib := kibConn.([]interface{})[0]; kib != nil {
+		kibConfig := kib.(map[string]interface{})
+
+		if useEnvAsDefault {
+			if username := os.Getenv("KIBANA_USERNAME"); username != "" {
+				config.Username = strings.TrimSpace(username)
+			}
+			if password := os.Getenv("KIBANA_PASSWORD"); password != "" {
+				config.Password = strings.TrimSpace(password)
+			}
+		}
+
+		if username, ok := kibConfig["username"]; ok {
+			config.Username = username.(string)
+		}
+		if password, ok := kibConfig["password"]; ok {
+			config.Password = password.(string)
+		}
+
+		if endpoints, ok := kibConfig["endpoints"]; ok && len(endpoints.([]interface{})) > 0 {
+			// We're curently limited by the API to a single endpoint
+			if endpoint := endpoints.([]interface{})[0]; endpoint != nil {
+				config.Address = endpoint.(string)
+			}
+		}
+
+		if insecure, ok := kibConfig["insecure"]; ok && insecure.(bool) {
+			config.DisableVerifySSL = true
+		}
+	}
+
+	kib, err := kibana.NewClient(config)
+
+	if err != nil {
+		diags = append(diags, diag.Diagnostic{
+			Severity: diag.Error,
+			Summary:  "Unable to create Kibana client",
+			Detail:   err.Error(),
+		})
+	}
+
+	return kib, diags
+}
+
+func newApiClient(d *schema.ResourceData, version string, useEnvAsDefault bool, esKey string) (*ApiClient, diag.Diagnostics) {
+	var diags diag.Diagnostics
+	baseConfig := buildBaseConfig(d, version, useEnvAsDefault)
```

Comment:
This will do the wrong thing when it's used against `elasticsearch_connection`. That'll have to change with the previous comment but we need to make sure we build the base config from the correct config block.